File: src/commands/tfti.rs
use poise::serenity_prelude::{Color, CreateEmbed, GetMessages};
use poise::{command, CreateReply};

type Error = Box<dyn std::error::Error + Send + Sync>;
type Context<'a> = poise::Context<'a, crate::Data, crate::Error>;

#[command(
  slash_command,
  description_localized("en-US", "Thanks for the invite, asshole")
)]
pub async fn tfti(ctx: Context<'_>) -> Result<(), Error> {
  let tfti_multiplier = ctx
    .channel_id()
    .messages(ctx.http(), GetMessages::new().limit(18))
    .await?
    .iter()
    .take_while(|message| {
      let is_bot_message = message.author.id == ctx.framework().bot_id;
      let is_tfti = message.embeds.iter().any(|embed| match &embed.title {
        Some(title) => title.starts_with("Tfti"),
        None => false,
      });
      is_bot_message && is_tfti
    })
    .count();

  let embed = CreateEmbed::default();
  let embed = match tfti_multiplier {
        0 => embed
            .title("Tfti")
            .description("Thanks for the invite, asshole.")
            .color(Color::from_rgb(243, 139, 168)),
        1 => embed
            .title("Tfti x2")
            .description("Thanks for the invite, asshole. Oh wait, that was just said.")
            .color(Color::from_rgb(250, 179, 135)),
        2 => embed
            .title("Tfti x3")
            .description("Thanks for the invite, asshole. Feels like I'm on repeat here.")
            .color(Color::from_rgb(166, 227, 161)),
        3 => embed
            .title("Tfti x4")
            .description("Thanks for the invite, asshole. Guess we'll just keep saying it.")
            .color(Color::from_rgb(137, 220, 235)),
        4 => embed
            .title("Tfti x5")
            .description("Thanks for the invite, asshole. I could stop, but why bother?")
            .color(Color::from_rgb(180, 190, 254)),
        5 => embed
            .title("Tfti x6")
            .description("Thanks for the invite, asshole. Just keep pretending I don't exist.")
            .color(Color::from_rgb(203, 166, 247)),
        6 => embed
            .title("Tfti x7")
            .description("Thanks for the invite, asshole. Still feels worth repeating.")
            .color(Color::from_rgb(245, 194, 231)),
        7 => embed
            .title("Tfti x8")
            .description("Thanks for the invite, asshole. I guess this is just what I do now.")
            .color(Color::from_rgb(148, 226, 213)),
        8 => embed.title("Tfti x9").description(
            "Thanks for the invite, asshole. Forever and always, from the bottom of my heart.")
            .color(Color::from_rgb(245, 224, 220)),
        _ => embed
            .title("Tfti ♾️")
            .description("Thanks for the invite, asshole. Forever stuck in this loop, I guess.")
            .color(Color::from_rgb(116, 199, 236)),
    };

  let builder = CreateReply::default().embed(embed);
  ctx.send(builder).await?;
  Ok(())
}

File: src/commands/tfti.ts
import { SlashCommandBuilder, ChatInputCommandInteraction, EmbedBuilder } from 'discord.js'
import { logger } from '../utils/logger'
import { SlashCommand } from '../models/commands'
import { ColorProvider } from '../utils/color-provider'

export const createTftiCommand = (colorProvider: ColorProvider): SlashCommand => ({
  command: new SlashCommandBuilder()
    .setName('tfti')
    .setDescription('Thanks for the invite, asshole'),

  async execute (interaction: ChatInputCommandInteraction) {
    try {
      const messages = await interaction.channel?.messages.fetch({ limit: 10 })
      const embed = new EmbedBuilder()
        .setURL('https://youtube.com/shorts/pFmq2xu8Hvw?si=ysapcGMaM6YqEcOI')
        .setColor(colorProvider.getRandomColor())

      if (!messages) {
        embed
          .setTitle('😤 Tfti')
          .setDescription('Thanks for the invite, asshole')
        await interaction.reply({ embeds: [embed] })
        return
      }

      const tftiMultiplier = [...messages.values()].map(message => {
        const isBotMessage = message.author.id === interaction.client.user?.id
        const isTfti = message.embeds.some(embed =>
          embed.title?.startsWith('😤 Tfti')
        )
        return isBotMessage && isTfti
      }).findIndex(bool => bool === false)

      switch (tftiMultiplier) {
        case 0:
          embed
            .setTitle('😤 Tfti')
            .setDescription('Thanks for the invite, asshole')
          break
        case 1:
          embed
            .setTitle('😤 Tfti x2')
            .setDescription('Thanks for the invite, asshole\nOh wait, that was just said')
          break
        case 2:
          embed
            .setTitle('😤 Tfti x3')
            .setDescription('Thanks for the invite, asshole\nFeels like I\'m on repeat here')
          break
        case 3:
          embed
            .setTitle('😤 Tfti x4')
            .setDescription('Thanks for the invite, asshole\nGuess we\'ll just keep saying it')
          break
        case 4:
          embed
            .setTitle('😤 Tfti x5')
            .setDescription('Thanks for the invite, asshole\nI could stop, but why bother?')
          break
        case 5:
          embed
            .setTitle('😤 Tfti x6')
            .setDescription('Thanks for the invite, asshole\nJust keep pretending I don\'t exist, as usual')
          break
        case 6:
          embed
            .setTitle('😤 Tfti x7')
            .setDescription('Thanks for the invite, asshole\nStill feels worth repeating')
          break
        case 7:
          embed
            .setTitle('😤 Tfti x8')
            .setDescription('Thanks for the invite, asshole\nI guess this is just what I do now')
          break
        case 8:
          embed
            .setTitle('😤 Tfti x9')
            .setDescription('Thanks for the invite, asshole\nForever and always, from the bottom of my heart')
          break
        default:
          embed
            .setTitle('😤 Tfti ♾️')
            .setDescription('Thanks for the invite, asshole\nTo infinity and beyond')
          break
      }

      await interaction.reply({ embeds: [embed] })
    } catch (error) {
      logger.error({ error }, 'Error in tfti command. Returning default tfti message.')
      const embed = new EmbedBuilder()
        .setURL('https://youtube.com/shorts/pFmq2xu8Hvw?si=ysapcGMaM6YqEcOI')
        .setTitle('😤 Tfti')
        .setDescription('Thanks for the invite, asshole')
        .setColor(colorProvider.getRandomColor())
      await interaction.reply({ embeds: [embed] })
    }
  }
})
